core/trino-main/src/main/java/io/trino/server/ui/OAuth2WebUiAuthenticationFilter.java

```diff
@@ -113,4 +120,22 @@ private void needAuthentication(ContainerRequestContext request)
         URI redirectLocation = service.startChallenge(request.getUriInfo().getBaseUri().resolve(CALLBACK_ENDPOINT));
         request.abortWith(Response.seeOther(redirectLocation).build());
     }
+
+    private boolean hasValidAudience(Claims claims)
+    {
+        if (validAudience.isEmpty()) {
+            return true;
```

Comment:
empty should mean that token cannot be used anywhere.

Reply:
My assumption was that when the valid audience is empty then the audience should not be enforced and any token will do. Having optional configuration property for which empty value means that no token will be valid makes little sense to me.